source/_posts/2017-02-07-falsehoods-programmers-believe-about-language.markdown

```diff
@@ -0,0 +1,189 @@
+should probably be solving other problems with your time.
+
+
+## Well, templating is bad, but if I allow for two versions of every English sentence, I'll be OK for gendered languages.
+
+This is wrong in two ways. First, two and one are not the only options for
+number of grammatical genders languages have. Some languages have masculine,
+feminine and _neuter_ nouns like German, Icelandic, Russian, and Polish. And
+some languages (with, I'll admit, reletively smaller speaking communities) have
+even more like Czech, which has four. Wikipedia, of course, has a
+[big, categorized list](https://en.wikipedia.org/wiki/List_of_languages_by_type_of_grammatical_genders).
+
+Second, if you have more than a few spots of semantic variability in a given
+sentence, you may run into a combinatorial explosion just going from English to
+a language with two grammatical genders. Similar to the above example about
+clothing color, consider `"Your friend sees your teacher."` This is sort of
+contrived to keep the other grammar simple, but could be any sentence with two
+titles for nouns in it.
+
+Spanish wants to know if your friend is gammatically male (_amigo_) or female
+(_amiga_) and then wants to know the same thing about your teacher (_profesor_
+or _profesora_). So you go from having one English sentence ("Your friend sees
+your teacher.") to four in Spanish:
+
+* Tu amigo ve a tu profesor.
+* Tu amiga ve a tu profesor.
+* Tu amigo ve a tu profesora.
+* Tu amiga ve a tu profesora.
+
+
+## At least templating users' names in is safe.
+
+It is my understanding that in Russian, which I do not speak or know much about,
+a verb has to agree in gender with the subject of the sentence, _only if it is
+past tense_. For those who haven't thought about a sentence diagram since the
+7th grade or whatever: The word for the action being taken has to agree in
+gender with the person taking the action, but only if the action is in the past.
+
+So something like, `"#{user.name} liked your post."` becomes fraught because
+"liked" is translated differently depending on the grammatical gender of the
+person who liked your post.
+
+
+## I can reword my copy to avoid these kinds of pitfalls.
+
+I really only speak English. I can sort of stumble my way around being
+understood in Spanish and I know a few random phrases in French and German and
+assortment of other languages. I know the first line of the _Ozymandias_
+pedestal in [Sindarin](https://en.wikipedia.org/wiki/Sindarin) becauase of
+course I do. Learning a language is hard, especially if you're interested in
+learning it well enough to know which sentence structures are going to be harder
+or easier to map back and forth to English.
+
+Even assuming you learned all 11 or 14 or howevermany languages your application
+is targeting, imagine trying to find the safe points in that 11- or
+14-dimensional grammatical graph. That sounds hard or maybe impossible to me.
```

Comment:
Even if you trained a neural network to find these spots... you'd then have to give control of writing all your copy to it, so it could exist in the safe spots, right? I think Google's thing is more about translating arbitrary (not "safe") languages between them in an automated way, which honestly seems like an easier problem to solve with better results, anyway.